src/components/AccountsPage/AccountList/index.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+
+import React, { Component } from 'react'
+import type { Account, PortfolioRange } from '@ledgerhq/live-common/lib/types'
+
+import Box from 'components/base/Box'
+import AccountListHeader from './Header'
+import GridBody from '../AccountList/GridBody'
+import ListBody from '../AccountList/ListBody'
+
+type Props = {
+  accounts: Account[],
+  onRangeChange: PortfolioRange => void,
+  onAccountClick: Account => void,
+  range: PortfolioRange,
+}
+
+type State = {
+  mode: string,
+  search: string,
+}
+
+class AccountList extends Component<Props, State> {
+  state = {
+    mode: 'card',
+    search: '',
+  }
+
+  onModeChange = () =>
+    this.setState(prevState => ({ mode: prevState.mode === 'card' ? 'list' : 'card' }))
+
+  onTextChange = (evt: SyntheticInputEvent<HTMLInputElement>) =>
+    this.setState({
+      search: evt.target.value,
+    })
+
+  render() {
+    const { accounts, range, onAccountClick, onRangeChange } = this.props
+    const { mode, search } = this.state
+    const isUsingCards = this.state.mode === 'card'
+    const Body = isUsingCards ? GridBody : ListBody
+
+    const filteredAccounts = accounts.filter(account =>
```

Comment:
for performance. we need to virtually split into two groups. the visibleAccounts and invisibleAccounts.

then in the rendering we will display the rows and the grid with this order: `[ ...visibles, placeholder, ...invisibles ]` (pseudo code). invisibles would be mapped with an extra "invisible". just make sure everything are still siblings, DOM speaking. that way, react will know that to optimize it, it just need to move DOM elements around.

Reply:
mmm I didn't notice any lag on my end, but I like this idea of only making them invisible instead of removing them, I'll play with it

Reply:
if you have 20+ accounts and you fastly play with the search switching between 'a' and '' , you notice it, it's slight but still affect the performance perception when you type characters. i locally fixed it on my end and the result was much better

Reply:
Solved with your pr